src/objects/Particle.cpp

```diff
@@ -1,56 +1,75 @@
 #include "Particle.h"
-
 #include "utils/ArrayUtils.h"
+#include "utils/CLIUtils.h"
 #include <iostream>
+#include <spdlog/spdlog.h>
+#include <string>
+#define MASS_ERROR "The mass of a particle must be positive for the currently available simulations!"
 
-Particle::Particle(int type_arg) {
-    type = type_arg;
-    f = {0., 0., 0.};
-    old_f = {0., 0., 0.};
-    std::cout << "Particle " << x << ", " << v << ", " << f << ", " << m << " generated (simple constructor)!"
-              << std::endl;
+Particle::Particle(int type_arg)
+    : x{0., 0., 0.}, v{0., 0., 0.}, f{0., 0., 0.}, old_f{0., 0., 0.}, m{1.}, type{type_arg} {
+    SPDLOG_TRACE("Generated Particle (simple constructor) - x: {}, v: {}, f: {}, m: {}", ArrayUtils::to_string(x),
+                 ArrayUtils::to_string(v), ArrayUtils::to_string(f), m);
+    if (m <= 0)
+        CLIUtils::error(MASS_ERROR);
 }
 
 Particle::Particle(const Particle &other)
     : x{other.x}, v{other.v}, f{other.f}, old_f{other.old_f}, m{other.m}, type{other.type} {
-    std::cout << "Particle " << x << ", " << v << ", " << f << ", " << m << " generated BY COPY!" << std::endl;
+    SPDLOG_TRACE("Generated Particle (copy) - x: {}, v: {}, f: {}, m: {}", ArrayUtils::to_string(x),
+                 ArrayUtils::to_string(v), ArrayUtils::to_string(f), m);
 }
 
 Particle::Particle(const std::array<double, 3> &x_arg, const std::array<double, 3> &v_arg, double m_arg, int type_arg)
     : x{x_arg}, v{v_arg}, f{0., 0., 0.}, old_f{0., 0., 0.}, m{m_arg}, type{type_arg} {
-    std::cout << "Particle " << x << ", " << v << ", " << f << ", " << m << " generated!" << std::endl;
+    SPDLOG_TRACE("Generated Particle (arguments) - x: {}, v: {}, f: {}, m: {}", ArrayUtils::to_string(x),
+                 ArrayUtils::to_string(v), ArrayUtils::to_string(f), m);
+    if (m <= 0)
+        CLIUtils::error(MASS_ERROR);
 }
 
 Particle::~Particle() {
-    std::cout << "Particle " << x << ", " << v << ", " << f << ", " << m << " destructed!" << std::endl;
+    SPDLOG_TRACE("Destroyed Particle - x: {}, v: {}, f: {}, m: {}", ArrayUtils::to_string(x), ArrayUtils::to_string(v),
+                 ArrayUtils::to_string(f), m);
 }
 
 const std::array<double, 3> &Particle::getX() const { return x; }
 const std::array<double, 3> &Particle::getV() const { return v; }
 const std::array<double, 3> &Particle::getF() const { return f; }
 const std::array<double, 3> &Particle::getOldF() const { return old_f; }
-const double Particle::getM() const { return m; }
-const int Particle::getType() const { return type; }
+double Particle::getM() const { return m; }
+int Particle::getType() const { return type; }

```

Comment:
If you are referring to why I used `double` instead of `const double`, I'm honestly not entirely sure myself. I think I vaguely remember setting them to / leaving them as `const`, then getting some sort of compiler warning or formatting error that got fixed when I removed `const`. Now it seems to work though, so they're `const`. I also wasn't all that familiar with the C++ conventions.